### ISSUE TITLE
wasmcli does not works.

### ISSUE THREAD

**Issue by westaking** [cosmwasm/wasmd#31]
## Summary of Bug

I can't generate a key with the wasmcli command.

## Version

os : centos 7
git commit hash: ec7f996

$ wasmd version --long
name: wasm
server_name: wasmd
client_name: wasmcli
version: ""
commit: ec7f9962182e98ee835dd63515828d77ed4eb336
build_tags: netgo,ledger
go: go version go1.13.6 linux/amd64

## Steps to Reproduce

`$ wasmcli keys add validator`

following this error:
```
panic: No such interface 'org.freedesktop.DBus.Properties' on object at path /

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/crypto/keys.keyringKeybase.writeInfo(0x13b2560, 0xc0003522a0, 0x7ffdbe8ea89d, 0x1, 0x13b25e0, 0xc00035a810)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/crypto/keys/keyring.go:444 +0x38c
github.com/cosmos/cosmos-sdk/crypto/keys.keyringKeybase.writeLocalKey(0x13b2560, 0xc0003522a0, 0x7ffdbe8ea89d, 0x1, 0x13b0860, 0xc000b38480, 0x105d674, 0x8, 0xc000b38480, 0xf73380)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/crypto/keys/keyring.go:430 +0x132
github.com/cosmos/cosmos-sdk/crypto/keys.baseKeybase.persistDerivedKey(0x13a2e20, 0xc000162b50, 0xc00035e380, 0x40, 0x40, 0x105d674, 0x8, 0x7ffdbe8ea89d, 0x1, 0xc000c07a90, ...)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/crypto/keys/keybase_base.go:104 +0x155
github.com/cosmos/cosmos-sdk/crypto/keys.baseKeybase.Derive(0x13a2e20, 0xc000162b50, 0x7ffdbe8ea89d, 0x1, 0xc000368140, 0x91, 0x0, 0x0, 0x105d674, 0x8, ...)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/crypto/keys/keybase_base.go:178 +0x16d
github.com/cosmos/cosmos-sdk/crypto/keys.baseKeybase.CreateAccount(0x13a2e20, 0xc000162b50, 0x7ffdbe8ea89d, 0x1, 0xc000368140, 0x91, 0x0, 0x0, 0x105d674, 0x8, ...)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/crypto/keys/keybase_base.go:88 +0x127
github.com/cosmos/cosmos-sdk/crypto/keys.keyringKeybase.CreateAccount(...)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/crypto/keys/keyring.go:78
github.com/cosmos/cosmos-sdk/client/keys.runAddCmd(0xc000334280, 0xc00050eff0, 0x1, 0x1, 0x0, 0x0)
        /home/user777/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20191114141721-d4c831e63ad3/client/keys/add.go:234 +0x91b
github.com/spf13/cobra.(*Command).execute(0xc000334280, 0xc00050efc0, 0x1, 0x1, 0xc000334280, 0xc00050efc0)
        /home/user777/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc000528500, 0xb9ee40, 0xc000528500, 0x1058d6c)
        /home/user777/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /home/user777/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/tendermint/tendermint/libs/cli.Executor.Execute(0xc000528500, 0x12549d0, 0x2, 0xc000b0d740)
        /home/user777/go/pkg/mod/github.com/tendermint/tendermint@v0.32.7/libs/cli/setup.go:89 +0x3c
main.main()
        /home/user777/wasmd/cmd/wasmcli/main.go:77 +0x5d5

```

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ethanfrey** [comment]
This code is straight from https://github.com/cosmos/cosmos-sdk.

Please file a bug with them about CentOS7 support. You can tag @alessio who probably knows more about the keyring keybase than anyone else on that team.

**Comment by ethanfrey** [comment]
Assigned "won't fix" as I cannot do anything until upstream support is added. Then I will just update the dependencies (so not closing it - please add a comment if anything changes)

**Comment by creamjun** [comment]
In centos7, other platforms based on tendermint can generate keys without problems.
However, wasmcli occurs an error on generating the key.

**Comment by westaking** [comment]
In centos7 other platforms based on tendermint can generate keys without problems. However, wasmcli occurs an error on generating the key.

**Comment by ethanfrey** [comment]
It's not based on tendermint - that works. It is the client library.

Which as of the recent master no longer uses a file system storage, but rather os-specific keyring.

To isolate the problem, can you please check out the current master branch of https://github.com/cosmos/gaia and `make install`. Then try the `gaiacli keys add validator` command (and all the other `keys` subcommands). If there is a bug, please log it on gaia and cosmos-sdk. If this works perfectly, I can look at it.

I am running on a very recent master, looking to tag to unreleased 0.38. This is much newer than most projects... recent cosmos hub upgrade was to 0.36. Some of the more adventurous projects are on 0.37.5 or so. I don't think anyone is running a net on master... I am waiting for the tagged release to do so as well.

**Comment by westaking** [comment]
yep, it does not work in the gaia master branch as well.  (cosmos-sdk v0.34.4)

**Comment by ethanfrey** [comment]
`cosmos-sdk v0.34.4` that is totally misleading tag (and has confused me many times). It is followed by a git commit, which you can search in cosmos-sdk for.

Can you please raise an issue on gaia and/or cosmos-sdk and link to this? Thank you

**Comment by westaking** [comment]
If I build directly with cosmos-sdk v0.34.4, generating key with cli works fine.
I'm confused too.

**Comment by ethanfrey** [comment]
Okay, I checked in with @alessio and there is a more recent PR that adds support for a file backend, which should work on more systems (like CentOS7 or Docker). https://github.com/cosmos/cosmos-sdk/pull/5355

It was merged Dec 11, which is newer than the version of cosmos-sdk we are using now. When we update to the current master #15 this flag will be enabled, and we can try it out. Hopefully that will resolve the problem.

(#15 is high priority for the release I plan later this week)

**Comment by westaking** [comment]
It seems that keyring does not support headless server.

So I saw the following article and solved it.

https://ahelpme.com/linux/dbusexception-could-not-get-owner-of-name-org-freedesktop-secrets-no-such-name/

**Comment by ethanfrey** [comment]
I'm glad it worked.
Do you want to make a PR adding that link and a little text to the README?

I'd be glad to have your contribution.

**Comment by westaking** [comment]
updated README.md  #40

**Comment by ethanfrey** [comment]
Closed by #40
